troposphere/static/js/components/projects/detail/resources/InstanceBatchActionButtons.jsx

```diff
@@ -0,0 +1,169 @@
+import React from "react";
+import Backbone from "backbone";
+// NOTE raven-js included during evaluation
+// - may be removed after ATMO-981 resolved
+import Raven from "raven-js";
+
+import Button from "./Button";
+import modals from "modals";
+
+
+/**
+ * I have some concern that calls to modals will fail,
+ * this allows us to capture them in Sentry
+ */
+function captureMsg(ex) {
+    if (Raven && Raven.isSetup()) {
+        if (Raven.captureException) {
+            Raven.captureException(ex);
+        }
+    }
+}
+
+
+export default React.createClass({
+    displayName: "InstanceBatchActionButtons",
+
+    propTypes: {
+        multipleSelected: React.PropTypes.bool.isRequired,
+        selectedResources: React.PropTypes.instanceOf(Backbone.Collection),
+        instance: React.PropTypes.instanceOf(Backbone.Model).isRequired,
+        project: React.PropTypes.instanceOf(Backbone.Model).isRequired
+    },
+
+    onStart: function() {
+        try {
+            modals.InstanceModals.start(this.props.selectedResources);
+            this.props.onUnselectAll();
+        } catch (ex) { captureMsg(ex); }
+    },
+
+    onSuspend: function() {
+        try {
+            modals.InstanceModals.suspend(this.props.selectedResources);
+            this.props.onUnselectAll();
+        } catch (ex) { captureMsg(ex); }
+    },
+
+    onStop: function() {
+        try {
+            modals.InstanceModals.stop(this.props.selectedResources);
+            this.props.onUnselectAll();
+        } catch (ex) { captureMsg(ex); }
+    },
+
+    onResume: function() {
+        try {
+            modals.InstanceModals.resume(this.props.selectedResources);
+            this.props.onUnselectAll();
+        } catch (ex) { captureMsg(ex); }
```

Comment:
@steve-gregory - so I think we can generalize, but not the way you are suggesting here. How the modal carries out its' duties via `actions` is exposed here, and I think we would aim for the component to still say `modals.InstanceModals.resume(this.props.selectedResources)` or maybe have the modal return a `Promise`: 

```
modals.InstanceModals
     .resume(this.props.selectedResources)
     .then(() => this.prop.onUnselectAll());
```

But, the assembling of actions with a function should be encapsulated by the modal or it is serving no purpose to the component calling it. 

That would look a bit too much like this:
![c3po-episode1](https://cloud.githubusercontent.com/assets/5923/22595932/b4cb526a-e9e6-11e6-8ecb-fb2b541071fc.jpg)